src/new/wallet.js

```diff
@@ -1008,30 +1178,165 @@ class HathorWallet extends EventEmitter {
    *  {
    *   'depositAddress': address of the HTR deposit back
    *   'changeAddress': address of the change output
+   *   'createAnotherMelt': boolean to create another melt authority or not for the wallet
+   *   'startMiningTx': boolean to trigger start mining (default true)
    *  }
    *
    * @return {Object} Object with {success: true, sendTransaction, promise}, where sendTransaction is a
    * SendTransaction object that emit events while the tx is being sent and promise resolves when the sending is done
    **/
-  meltTokens(tokenUid, amount, options = { depositAddress: null, changeAddress: null }) {
+  meltTokens(tokenUid, amount, options = {}) {
     storage.setStore(this.store);
+    const newOptions = Object.assign({
+      depositAddress: null,
+      changeAddress: null,
+      createAnotherMelt: true,
+      startMiningTx: true,
+    }, options);
     const meltInput = this.selectAuthorityUtxo(tokenUid, wallet.isMeltOutput.bind(wallet));
 
-    if (meltInput === null) {
+    if (meltInput.length === 0) {
       return {success: false, message: 'Don\'t have melt authority output available.'}
     }
 
     // Always create another melt authority output
-    const ret = tokens.meltTokens(meltInput, tokenUid, amount, this.pinCode, true, options);
+    const ret = tokens.meltTokens(meltInput[0], tokenUid, amount, this.pinCode, newOptions.createAnotherMelt, newOptions);
     if (ret.success) {
       const sendTransaction = ret.sendTransaction;
-      sendTransaction.start();
+      if (newOptions.startMiningTx) {
+        sendTransaction.start();
+      }
+      return {success: true, promise: sendTransaction.promise, sendTransaction};
+    } else {
+      return ret;
+    }
+  }
+
+  /**
+   * Delegate authority
+   *
+   * @param {String} tokenUid UID of the token to delegate the authority
+   * @param {String} type Type of the authority to delegate 'mint' or 'melt'
+   * @param {String} destinationAddress Destination address of the delegated authority
+   * @param {Object} options Options parameters
+   *  {
+   *   'createAnother': if should create another authority for the wallet. Default to true
+   *   'startMiningTx': boolean to trigger start mining (default true)
+   *  }
+   *
+   * @return {Object} Object with {success: true, sendTransaction, promise}, where sendTransaction is a
+   * SendTransaction object that emit events while the tx is being sent and promise resolves when the sending is done
+   **/
+  delegateAuthority(tokenUid, type, destinationAddress, options = {}) {
+    storage.setStore(this.store);
```

Comment:
Why do we have to set the store on every method?

Reply:
The wallet headless might have many wallets handled by the same lib. Even though the wallets have separated storages and the class is prepared for that, the lib itself is not, thus we access the libStore many times.

For that to work with async calls we need to call `setStore` every time to indicates which store we want to use in each call.